### PR TITLE
Adding onDropped callback to throttleLast as a part of #7458

### PR DESCRIPTION
**Pull request by Desislav-Petrov**
Adding onDropped callback for throttleLast - #7458

### CODE REVIEW

File: src/main/java/io/reactivex/rxjava3/core/Flowable.java [hunk 1]
```diff
@@ -14713,7 +14820,51 @@ public final Flowable<T> sample(long period, @NonNull TimeUnit unit, @NonNull Sc
...
+     * @see <a href="http://reactivex.io/documentation/operators/sample.html">ReactiveX operators documentation: Sample</a>
+     * @see <a href="https://github.com/ReactiveX/RxJava/wiki/Backpressure">RxJava wiki: Backpressure</a>
+     * @see #throttleLast(long, TimeUnit, Scheduler)
+     * @since 2.1
+     */
+    @CheckReturnValue
+    @NonNull
+    @BackpressureSupport(BackpressureKind.ERROR)
+    @SchedulerSupport(SchedulerSupport.CUSTOM)
+    public final Flowable<T> sample(long period, @NonNull TimeUnit unit, @NonNull Scheduler scheduler, boolean emitLast, @NonNull Consumer<T> onDropped) {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/rxjava3/core/Flowable.java]: Please only add this overload to `sample` and not the others.

File: src/main/java/io/reactivex/rxjava3/core/Flowable.java [hunk 2]
```diff
@@ -17211,6 +17398,45 @@ public final Flowable<T> throttleLast(long intervalDuration, @NonNull TimeUnit u
...
+     * @throws NullPointerException if {@code unit} or {@code scheduler} is {@code null} or {@code onDropped} is {@code null}
+     * @see <a href="http://reactivex.io/documentation/operators/sample.html">ReactiveX operators documentation: Sample</a>
+     * @see <a href="https://github.com/ReactiveX/RxJava/wiki/Backpressure">RxJava wiki: Backpressure</a>
+     * @see #sample(long, TimeUnit, Scheduler)
+     */
+    @CheckReturnValue
+    @BackpressureSupport(BackpressureKind.ERROR)
+    @SchedulerSupport(SchedulerSupport.CUSTOM)
+    @NonNull
+    public final Flowable<T> throttleLast(long intervalDuration, @NonNull TimeUnit unit, @NonNull Scheduler scheduler, @NonNull Consumer<T> onDropped) {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/rxjava3/core/Flowable.java]: Please only add this `throttleLast` overload and no other.

File: src/main/java/io/reactivex/rxjava3/core/Observable.java [hunk 3]
```diff
@@ -12128,7 +12190,40 @@ public final Observable<T> sample(long period, @NonNull TimeUnit unit, boolean e
...
+     *          called with the current entry when it has been replaced by a new one
+     * @return the new {@code Observable} instance
+     * @throws NullPointerException if {@code unit} or {@code scheduler} is {@code null} or {@code onDropped} is {@code null}
+     * @see <a href="http://reactivex.io/documentation/operators/sample.html">ReactiveX operators documentation: Sample</a>
+     * @see #throttleLast(long, TimeUnit, Scheduler)
+     */
+    @CheckReturnValue
+    @SchedulerSupport(SchedulerSupport.CUSTOM)
+    @NonNull
+    public final Observable<T> sample(long period, @NonNull TimeUnit unit, @NonNull Scheduler scheduler, @NonNull Consumer<T> onDropped) {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/rxjava3/core/Observable.java:12160]: Please only add this `sample` overload.

File: src/main/java/io/reactivex/rxjava3/core/Observable.java [hunk 4]
```diff
@@ -14233,6 +14368,73 @@ public final Observable<T> throttleLast(long intervalDuration, @NonNull TimeUnit
...
+     *          called with the current entry when it has been replaced by a new one
+     * @return the new {@code Observable} instance
+     * @throws NullPointerException if {@code unit} or {@code scheduler} is {@code null} or {@code onDropped} is {@code null}
+     * @see <a href="http://reactivex.io/documentation/operators/sample.html">ReactiveX operators documentation: Sample</a>
+     * @see #sample(long, TimeUnit, Scheduler)
+     */
+    @CheckReturnValue
+    @SchedulerSupport(SchedulerSupport.CUSTOM)
+    @NonNull
+    public final Observable<T> throttleLast(long intervalDuration, @NonNull TimeUnit unit, @NonNull Scheduler scheduler, @NonNull Consumer<T> onDropped) {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/rxjava3/core/Observable.java:14300]: Please only add this `throttleLast` overload.

File: src/main/java/io/reactivex/rxjava3/core/Observable.java [hunk 5]
```diff
@@ -14279,7 +14481,7 @@ public final Observable<T> throttleLast(long intervalDuration, @NonNull TimeUnit
      * If no items were emitted from the upstream during this timeout phase, the next
      * upstream item is emitted immediately and the timeout window starts from then.
      * <dl>
-     *  <dt><b>Scheduler:</b></dt>
+     *  <dt><b>Scheduler:</b></dt>&
```
Comment (akarnokd) [inline src/main/java/io/reactivex/rxjava3/core/Observable.java]: Unnecessary change.

**Review by akarnokd — CHANGES_REQUESTED** [review]

**Comment by Desislav-Petrov** [comment]
hi @akarnokd thx for the review - all addressed

**Review by akarnokd — APPROVED** [review]

**Pull request merged**